prow/spyglass/spyglass.go

```diff
@@ -85,7 +85,7 @@ func New(ctx context.Context, ja *jobs.JobAgent, cfg config.Getter, opener pkgio
 		JobAgent:               ja,
 		config:                 cfg,
 		PodLogArtifactFetcher:  NewPodLogArtifactFetcher(ja),
-		StorageArtifactFetcher: NewStorageArtifactFetcher(opener, useCookieAuth),
+		StorageArtifactFetcher: NewStorageArtifactFetcher(opener, cfg().ValidateStorageBucket, useCookieAuth),
```

Comment:
We should pass a dynamic config getter here rather than a method of the static config struct. As is, if a new job is added with a new bucket, we'd have to restart Deck in order for it to respect the newly valid bucket.

Reply:
Done.